src/Facebook/InstantArticles/Elements/Footer.php

```diff
@@ -189,4 +194,47 @@ public function toDOMElement($document = null)
 
         return $footer;
     }
+
+    /**
+     * Overrides the @see Element::isValid().
+     *
+     * @return true for valid Footer when it is filled, false otherwise.
+     */
+    public function isValid()
+    {
+        return
+            $this->credits ||
+            $this->copyright ||
+            $this->relatedArticles;
+    }
+
+    /**
+     * Implements the @see Container::getContainerChildren().
+     *
+     * @return array of Paragraph|RelatedArticles
+     */
+    public function getContainerChildren()
+    {
+        $children = array();
+
+        if ($this->credits) {
+            if (is_array($this->credits)) {
+                foreach ($this->credits as $paragraph) {
+                    if (Type::is($paragraph, Element::getClassName())) {
```

Comment:
Just noted that on:

https://github.com/facebook/facebook-instant-articles-sdk-php/blob/ed25a2209ff2019fec92d417566c06fc5125b7d5/src/Facebook/InstantArticles/Elements/Footer.php#L68

We are not enforcing the type of the array to be only `Paragraph`s. Please add this validation there.